sys/kern/interrupt.c

```diff
@@ -146,9 +169,12 @@ void intr_event_run_handlers(intr_event_t *ie) {
       if (ie->ie_disable)
         ie->ie_disable(ie);
 
-      TAILQ_REMOVE(&ie->ie_handlers, ih, ih_link);
-      TAILQ_INSERT_TAIL(&delegated, ih, ih_link);
-      sleepq_signal(&delegated);
+      if (ie->ie_ithread == NULL)
+        intr_thread_create(ie);
+
+      ih->ih_flags = IH_DELEGATE;
+      sleepq_signal(&ie->ie_ithread->it_wchan);
+
```

Comment:
Note that we can have more handlers within an interrupt event which require attention when `intr_event_run_handlers` is called. The problem is that this procedure isn't adapted to such a case. Please see #794 .